Test/Unit/Model/Api/ShippingMethodsTest.php

```diff
@@ -610,56 +722,439 @@ public function shippingEstimationWithDiscountAppliedToShipping()
             ->method('getGroupedAllShippingRates')
             ->willReturn($shippingRates);
 
+        $this->setupShippingOptionFactory(
+            "Flate Rate - Fixed [{$shippingDiscountAmount} discount]",
+            'flatrate_flatrate',
+            500, 0
+        );
+
         $quote = $this->getQuoteMock($shippingAddress);
 
         $this->setUpRuleFactoryMock();
-        $currentTestObject = $this->getCurrentTestObject();
+        $this->initCurrentMock();
 
-        $result = $currentTestObject->shippingEstimation($quote, $shippingAddressData);
+        $result = $this->currentMock->shippingEstimation($quote, $shippingAddressData);
 
         $this->assertEquals($this->factoryShippingOptionsMock, $result);
     }
 
     /**
      * @test
-     * @throws \ReflectionException
      */
-    public function testCouponInvalidForShippingAddress()
+    public function shippingEstimation_cached()
     {
-        $parentQuoteCoupon = 'IGNORED_SHIPPING_ADDRESS_COUPON';
-        $configCoupons = ['BOLT_TEST', 'ignored_shipping_address_coupon'];
-        $immutableQuoteMock = $this->getQuoteMock([]);
+        $email = "integration@bolt.com";
+        $shippingAddressData = [
+            'company'         => "",
+            'country'         => "United States",
+            'country_code'    => "US",
+            'email'           => $email,
+            'first_name'      => "John",
+            'last_name'       => "McCombs",
+            'locality'        => "Knoxville",
+            'phone'           => "+1 231 231 1234",
```

Comment:
The same as above regarding phone format.

Reply:
done